src/gui/curses/gui-curses-chat.c

```diff
@@ -1873,6 +1875,9 @@ gui_chat_draw_free_buffer (struct t_gui_window *window, int clear_chat)
             ptr_line = gui_line_get_next_displayed (ptr_line);
         if (ptr_line)
         {
+            window->scroll->first_line_displayed =
+                (ptr_line == gui_line_get_first_displayed (window->buffer));
+
```

Comment:
I fixed this problem in the issue #899, commit 5d4220a91f8aa1467302dac2e7a4bfdb7a017009, so the commit in this issue for the bare display scrolling issue could be removed.
By the way, I think your fix was not working properly: it always sets first_line_displayed to 0 because the variable should be set earlier (before display).